tests/network/vmi_istio.go

```diff
@@ -0,0 +1,297 @@
+					Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).NotTo(Succeed())
+				},
+					table.Entry(fmt.Sprintf("client outside mesh should not reach VMI HTTP server on service declared port %d", svcDeclaredTestPort), svcDeclaredTestPort),
+					table.Entry(fmt.Sprintf("client outside mesh should not reach VMI HTTP server on service undeclared port %d", svcUndeclaredTestPort), svcUndeclaredTestPort),
+				)
+			})
+		})
+		Describe("Outbound traffic", func() {
+			It("Should be able to reach kubevirt.io", func() {
+				// Istio envoy may intercept the request, we need to check the header response, not just the
+				// return code of curl, because if envoy proxy forbids a request, it responds with 502 Bad Gateway code
+				err = console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: "curl -sD - -o /dev/null http://kubevirt.io | head -n 1\n"},
+					&expect.BExp{R: fmt.Sprintf(".*%s.*\n", "HTTP/1.1 301 Moved Permanently")},
+					&expect.BSnd{S: "echo $?\n"},
+					&expect.BExp{R: console.RetValue("0")},
+				}, 60)
+				Expect(err).ToNot(HaveOccurred())
+			})
+			Context("With Sidecar allowing only registered external services", func() {
+				var registryOnlySidecar *istiogov1beta1.Sidecar
+				BeforeEach(func() {
+					virtClient, err = kubecli.GetKubevirtClient()
+					tests.PanicOnError(err)
+
+					registryOnlySidecar = &istiogov1beta1.Sidecar{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "default",
+							Namespace: tests.NamespaceTestDefault,
+						},
+						Spec: istiov1beta1.Sidecar{
+							OutboundTrafficPolicy: &istiov1beta1.OutboundTrafficPolicy{
+								Mode: istiov1beta1.OutboundTrafficPolicy_REGISTRY_ONLY,
+							},
+						},
+					}
+					_, err = virtClient.IstioClient().NetworkingV1beta1().Sidecars(tests.NamespaceTestDefault).Create(context.TODO(), registryOnlySidecar, metav1.CreateOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+				})
+				AfterEach(func() {
+					err = virtClient.IstioClient().NetworkingV1beta1().Sidecars(tests.NamespaceTestDefault).Delete(context.TODO(), registryOnlySidecar.Name, metav1.DeleteOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+				})
+				It("Should not be able to reach kubevirt.io", func() {
+					// Istio will intercept the request because of the OutboundTrafficPolicy set to REGISTRY_ONLY.
+					// Envoy will response with 502 Bad Gateway return code.
+					// After Sidecar with OutboundTrafficPolicy is created, it may take a while for the proxy
+					// to sync with the change, so first request may still pass.
+					Eventually(func() error {
+						return console.SafeExpectBatch(vmi, []expect.Batcher{
+							&expect.BSnd{S: "\n"},
+							&expect.BExp{R: console.PromptExpression},
+							&expect.BSnd{S: "curl -sD - -o /dev/null http://kubevirt.io | head -n 1\n"},
+							&expect.BExp{R: fmt.Sprintf(".*%s.*\n", "HTTP/1.1 502 Bad Gateway")},
```

Comment:
You could reuse the function mentioned above here.

Reply:
Done